### DISCUSSION TITLE
Not sure how to do this...

### DISCUSSION THREAD

**Discussion by thedotmack** [alpinejs/alpine#708] [4. General]
I am pretty new to this, and I'm digging in to Alpine for the first time right now.

I'm trying to toggle this via a custom event from another library, Vimeo's player library:

`<div x-data="{ showModal: false }">`

with this event

```
playerCommercial.on('fullscreenchange', function(data){ 
   // Toggle the modal here somehow, this is my issue
})
```

**Comment by SimoTod** [comment]
@thedotmack You can trigger event from vimeo and capture it in your alpine component.
```javascript 
playerCommercial.on('fullscreenchange', function(data) {
    document.dispatchEvent(new CustomEvent('open-modal' ))
})
```
```html
<div x-data="{ showModal: false }" @open-modal.document="showModal = true">
...
</div>
```

**Reply by thedotmack** [reply]
That's rad. Thank you!
